backend/dynamic-ui/http-data-modelling.ftd

```diff
@@ -0,0 +1,255 @@
+seemlessly connect the backend with the frontend.
+
+The first step is to create a `fastn package`.
+
+-- cbox.info: What is `fastn` package?
+
+`fastn` package is a folder that requires atleast two files
+
+- FASTN.ftd
+- index.ftd
+
+-- ds.h2: http processor
+
+`http` processor does the network communication using REST API and fetches data
+from the external site in a different domain.
+
+The syntax of using `http` processor is as follows:
+
+-- ds.code:
+
+\$processor$: pr.http
+
+
+-- ds.h2: Data modelling: `record`
+
+For this example, we are going to use `fastn`'s native type called `record`.
+`Record` is a user defined type that can be used to hold the country data.
+
+The syntax of record is as follows:
+
+-- ds.code: Declaring a country record
+
+\-- record country-data:
+string name:
+string capital:
+
+-- ds.h2: Start building
+
+Let's implement the theory and build the project. 
+
+-- ds.h3: **Step 1:** Import `fastn` processor
+
+Copy the import line and paste it at the top of the `index.ftd` document
+
+-- ds.code: Import Processors
+
+\-- import: fastn/processors as pr
+
+
+-- ds.h3: **Step 2:** Declare a `record` 
+
+Before a record can be used it must be declared
+
+
+-- ds.code: Declaring `country-data` record
+
+\-- record country-data:
+string name:
+string capital:
+
+
+-- ds.h3: **Step 3:** Create a list
+
+Since the JSON data has a list of countries and their respective captials, 
+therefore, we will create a list and use the `country-data` record as the type.
+
+-- ds.code: list countries of country type
+
+\-- country-data list countries:
+
+
+-- ds.markdown:
+
+This list will be created when we do the `http` calling with the `http`
+processor. Also, we will provide the endpoint i.e. the address where the data
+is present as the `url`.
+
+-- ds.code: 
+
+\-- country-data list countries:
+$processor$: pr.http
+url: https://famous-loincloth-ox.cyclic.app/
+
+
+
+-- ds.h3: **Step 4:** Create a component definition
+
+
+-- ds.code: Create component `country-detail`
+
+\-- component country-detail:
+
+
+\-- end: country-detail
+
+
+-- ds.markdown: 
+
+Create a header property `country` of type **country-data** as caption
+
+-- ds.code:
+
+\-- component country-detail:
+caption country-data country:  ;; <hl>
+
+\-- end: country-detail
+
+-- ds.markdown:
+
+Let's first pull the data of country and display in the browser
+
+-- ds.code:
+
+\-- component country-detail:
+caption country-data country:  
+
+\-- ftd.text: $country-detail.country.name  ;; <hl>
+
+\-- end: country-detail
+
+
+
+
+-- ds.markdown:
+
+Till now, we have just defined the component but to display the list of country
+names we need call the component. Therefore, after closing the component we 
+can call the component and create a `for` loop. 
+
+
+-- ds.code:
+
+\-- country-detail: $country
+for: $country in $countries
+
+-- ds.markdown:
+
+Now wrap the two texts for country-name and capital in `row` container
```

Comment:
What is `country-name`? I see country.name. And if you call name country-name, then why are you calling capital capital and not country-capital?